endpoints/openrtb2/amp_auction.go

```diff
@@ -139,6 +144,11 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 	w.Header().Set("Access-Control-Expose-Headers", "AMP-Access-Control-Allow-Source-Origin")
 	w.Header().Set("X-Prebid", version.BuildXPrebidHeader(version.Ver))
 
+	if _, reject := deps.hookExecutor.ExecuteEntrypointStage(r, nil); reject != nil {
```

Comment:
I recommend adding a comment explaining there is no body for AMP requests to explain the nil argument. Or perhaps use a private package scoped variable called `nilBody` set to nil and pass that in here.